### ISSUE TITLE
asr negative loss

### ISSUE THREAD

**Issue by yijingshihenxiule** [yl4579/StyleTTS2#236]
Hi， @yl4579 ,thank you for your awesome work.
I met some problems when train my model. When I train asr model with my phoneme symbols, I got negative ctc loss and negative train loss. Is it normal? Could you give me some suggestions?

-   5 INFO:2024-05-04 19:30:14,254: --- epoch 1 ---
  4 INFO:2024-05-04 19:30:14,254: train/loss     : 3.1812
  3 INFO:2024-05-04 19:30:14,254: train/ctc      : 0.6970
  2 INFO:2024-05-04 19:30:14,255: train/s2s      : 2.4842
  1 INFO:2024-05-04 19:30:14,255: train/learning_rate: 0.0005
6   INFO:2024-05-04 19:30:14,255: eval/ctc       : -0.8486
  1 INFO:2024-05-04 19:30:14,255: eval/s2s       : 1.7515
  2 INFO:2024-05-04 19:30:14,255: eval/loss      : 0.9029
  3 INFO:2024-05-04 19:30:14,256: eval/wer       : 0.1093
  4 INFO:2024-05-04 19:30:14,256: eval/acc       : 0.5668
  5 INFO:2024-05-04 19:41:44,964: --- epoch 2 ---
  6 INFO:2024-05-04 19:41:44,964: train/loss     : 0.5994
  7 INFO:2024-05-04 19:41:44,965: train/ctc      : -0.4791
  8 INFO:2024-05-04 19:41:44,965: train/s2s      : 1.0785
  9 INFO:2024-05-04 19:41:44,965: train/learning_rate: 0.0005
 10 INFO:2024-05-04 19:41:44,965: eval/ctc       : -1.0261
 11 INFO:2024-05-04 19:41:44,965: eval/s2s       : 0.3319
 12 INFO:2024-05-04 19:41:44,966: eval/loss      : -0.6943
 13 INFO:2024-05-04 19:41:44,966: eval/wer       : 0.0790
 14 INFO:2024-05-04 19:41:44,966: eval/acc       : 0.9172
 15 INFO:2024-05-04 19:53:16,502: --- epoch 3 ---
 16 INFO:2024-05-04 19:53:16,503: train/loss     : -0.0775
 17 INFO:2024-05-04 19:53:16,503: train/ctc      : -0.6186
 18 INFO:2024-05-04 19:53:16,503: train/s2s      : 0.5411
 19 INFO:2024-05-04 19:53:16,504: train/learning_rate: 0.0005
 20 INFO:2024-05-04 19:53:16,504: eval/ctc       : -1.0138
 21 INFO:2024-05-04 19:53:16,504: eval/s2s       : 0.2688
 22 INFO:2024-05-04 19:53:16,504: eval/loss      : -0.7450
 23 INFO:2024-05-04 19:53:16,505: eval/wer       : 0.1011
 24 INFO:2024-05-04 19:53:16,505: eval/acc       : 0.9133
 25 INFO:2024-05-04 20:04:42,985: --- epoch 4 ---
 26 INFO:2024-05-04 20:04:42,985: train/loss     : -0.2492
 27 INFO:2024-05-04 20:04:42,986: train/ctc      : -0.6826
 28 INFO:2024-05-04 20:04:42,986: train/s2s      : 0.4334
 29 INFO:2024-05-04 20:04:42,986: train/learning_rate: 0.0005
 30 INFO:2024-05-04 20:04:42,986: eval/ctc       : -1.0920
 31 INFO:2024-05-04 20:04:42,987: eval/s2s       : 0.2168
 32 INFO:2024-05-04 20:04:42,987: eval/loss      : -0.8752
 33 INFO:2024-05-04 20:04:42,987: eval/wer       : 0.0761
 34 INFO:2024-05-04 20:04:42,987: eval/acc       : 0.9345

**Comment by martinambrus** [comment]
Maybe you'll get more responses if you add this as [an ASR repository issue](https://github.com/yl4579/AuxiliaryASR/issues).

**Comment by ltphuongunited** [comment]
Hi @yijingshihenxiule, have you resolved this issue? I also have the same problem as you.
